lua/dylib.lua

```diff
@@ -30,6 +68,79 @@ int add_fib_entry(const char *prefix, const char *gt_ip, const char *gw_ip,
 	enum gk_fib_action action, struct gk_config *gk_conf);
 int del_fib_entry(const char *ip_prefix, struct gk_config *gk_conf);
 
+int convert_ip_to_str(struct ipaddr *ip_addr, char *res, int n);
+
+size_t strlen(const char *s);
+
 ]]
 
 c = ffi.C
+
+-- The following is an example function that can be used as
+-- the callback function of list_fib4_entries() and list_fib6_entries().
+
+-- Parameter fib_dump_entry is going to be released after
+-- print_fib_dump_entry() returns, so don't keep references to fib_dump_entry
+-- or any of the data reachable through its fields.
+
+function print_fib_dump_entry(fib_dump_entry)
+	local i = 0
+	local ret
+	local res = ""
+	local max_ip_addr_len = 128
+	local entry = ffi.cast("struct gk_fib_dump_entry *", fib_dump_entry)
+	local ip_addr = ffi.new("char[?]", max_ip_addr_len)
+	local ip_addr_str
+
+	ret = c.convert_ip_to_str(entry.addr, ip_addr, max_ip_addr_len)
```

Comment:
Add a function Lua C function that returns a Lua string of the proper size. Having to create variable `ip_addr` as `ffi.new("char[?]", max_ip_addr_len)` is adding unnecessary complexity here. We must keep the Lua scripts simple because operators will eventually need to edit them during stressful times.

To make my point even more clearer, `c.convert_ip_to_str()` is used three times in this function. Once we have a Lua C function to do the work, besides avoiding the need to allocate a buffer in Lua, the error test after the call of `c.convert_ip_to_str()` goes away since the error can be raised in the new function if needed, and there won't be a need to convert the buffer into a Lua string. That is much simpler code. The final result will be what you accomplished with `dylib.ether_format_addr()`.